src/Kros.AspNetCore/Authorization/GatewayAuthorizationMiddleware.cs

```diff
@@ -137,10 +145,36 @@ internal class GatewayAuthorizationMiddleware
 
                 string jwtToken = await client.GetStringAndCheckResponseAsync(authorizationUrl,
                     new UnauthorizedAccessException(Properties.Resources.AuthorizationServiceForbiddenRequest));
-                SetTokenToCache(memoryCache, cacheKey, jwtToken, httpContext.Request);
+                string memoryValue = GetMemoryValue(jwtToken, connectionId);
+                SetTokenToCache(memoryCache, cacheKey, memoryValue, httpContext.Request);
 
+                return memoryValue;
+            }
+        }
+
+        private string GetMemoryValue(string jwtToken, string connectionId)
+        {
+            if (string.IsNullOrWhiteSpace(connectionId))
+            {
                 return jwtToken;
             }
+            else
+            {
+                return $"{jwtToken} {ConnectionIdHelper.ConnectionId}:{connectionId}";
```

Comment:
Je toto potrebné? Síce sa to nejak ukladá do keše, ale potom sa s tým nijako nepracuje. Keď sa to z keše vytiahne, tak sa táto časť len odstrihne a zahodí. Mne to príde, že do keše stále stačí ukladať len pôvodný token a `ConnectionId` slúži len na vytváranie keš kľúča.

Reply:
nie je to potrebne, staci ak sa len key vygeneruje aj s cache http headers